src/commands/image/imageSource/buildImageInAzure/BuildImageStep.ts

```diff
@@ -3,60 +3,121 @@
-
-        if (outputImages) {
-            const image = outputImages[0];
-            context.image = `${image.registry}/${image.repository}:${image.tag}`;
-        } else {
-            const logSasUrl = (await context.client.runs.getLogSasUrl(context.resourceGroupName, context.registryName, nonNullValue(context.run.runId))).logLink;
-            const response: AzExtPipelineResponse = await sendRequestWithTimeout(context, { method: 'GET', url: nonNullValue(logSasUrl) }, 2500, undefined);
-            const content: string = nonNullProp(response, 'bodyAsText');
-
-            this.acrBuildError = {
-                name: nonNullValueAndProp(context.run, 'name'),
-                runId: nonNullValueAndProp(context.run, 'id'),
-                content
-            };
+    protected async executeCore(context: BuildImageInAzureImageSourceContext, progress: Progress<{ message?: string | undefined; increment?: number | undefined }>): Promise<void> {
+        const runRequest: AcrDockerBuildRequest = {
+            type: 'DockerBuildRequest',
+            imageNames: [context.imageName],
+            isPushEnabled: true,
+            sourceLocation: context.uploadedSourceLocation,
+            platform: { os: context.os },
+            dockerFilePath: path.relative(context.srcPath, context.dockerfilePath)
+        };
+
+        progress.report({ message: localize('buildingImage', 'Building image...') });
+        ext.outputChannel.appendLog(localize('buildingImageLog', 'Building image...'));
+
+        let acrRunId: string | undefined;
+        while (true) {
+            if (this.acrRunCount > 0) {
+                await delay(DELAY_BETWEEN_RUN_REQUESTS_MS);
 
-            const viewLogsButton: MessageItem = { title: localize('viewLogs', 'View Logs') };
-            const errorMessage = localize('noImagesBuilt', 'Failed to build image. View logs for more details.');
+                progress.report(({ message: localize('buildingImageRetry', 'Building image (Attempt {0}/{1})...', this.acrRunCount + 1, RETRY_LIMIT) }));
+                ext.outputChannel.appendLog(localize('buildingImageRetryLog', 'Building image (Attempt {0}/{1})...', this.acrRunCount + 1, RETRY_LIMIT));
+            }
 
-            void window.showErrorMessage(errorMessage, viewLogsButton).then(async result => {
-                if (result === viewLogsButton) {
-                    await openAcrBuildLogs(context, this.acrBuildError);
-                }
-            });
+            // Schedule the ACR build image run
+            try {
+                acrRunId = (await context.client.registries.beginScheduleRunAndWait(context.resourceGroupName, context.registryName, runRequest)).runId;
+                this.acrRunCount++;
+            } catch (e) {
+                // No known benefits to enforcing retries at this stage-- only check for retries if we can interpret specific actions from the build logs
+                await this.removeTarFile(context);
```

Comment:
Don't we want to remove this on success as well? I think we should put it in a `finally`.

Reply:
Yeah, I was kind of hoping to find a way to not have to introduce the mega try/finally wrapper 😂